native/server/src/sqlite_brongnal.rs

```diff
@@ -0,0 +1,350 @@
+
+        Ok(SqliteStorage(Arc::new(Mutex::new(connection))))
+    }
+}
+
+impl Storage for SqliteStorage {
+    fn add_user(
+        &self,
+        identity: String,
+        identity_key: VerifyingKey,
+        signed_pre_key: proto::service::SignedPreKey,
+    ) -> tonic::Result<()> {
+        eprintln!("Adding user: {identity}");
+
+        let _ = self.connection()?.execute(
+            "INSERT INTO user (identity, key, current_pre_key, creation_time) VALUES (?1, ?2, ?3, ?4)",
+            (
+                identity, identity_key.to_bytes(), signed_pre_key.encode_to_vec(),
+                SystemTime::now().duration_since(UNIX_EPOCH).unwrap().as_secs(),
+            ),
+        ).context("failed to insert key.");
+        Ok(())
+    }
+
+    fn clear_one_time_keys(&self, identity: &str) -> tonic::Result<()> {
+        eprintln!("Clearing one time keys for user: {identity}");
+        let _ = self
+            .connection()?
+            .execute(
+                "DELETE FROM pre_key WHERE user_identity = ?1",
+                params![identity.to_owned()],
+            )
+            .map_err(|e| {
+                Status::internal(format!("failed to clear one time keys for {identity}: {e}"))
+            })?;
+        Ok(())
+    }
+
+    fn update_pre_key(
+        &self,
+        identity: &str,
+        signed_pre_key: proto::service::SignedPreKey,
+    ) -> tonic::Result<()> {
+        eprintln!("Updating pre key for user: {identity}");
+
+        let _: String = self
+            .connection()?
+            .query_row(
+                "UPDATE user SET current_pre_key = ?2 WHERE identity = ?1 RETURNING identity",
+                params![identity, signed_pre_key.encode_to_vec()],
+                |row| Ok(row.get(0)?),
+            )
+            .map_err(|_| Status::not_found("user not found"))?;
+        Ok(())
+    }
+
+    fn add_one_time_keys(
+        &self,
+        identity: &str,
+        pre_keys: Vec<X25519PublicKey>,
+    ) -> tonic::Result<()> {
+        eprintln!(
+            "Adding {} one time keys for user: {identity}",
+            pre_keys.len()
+        );
+
+        let connection = self.connection()?;
+        let mut stmt = connection
+            .prepare("INSERT INTO pre_key (user_identity, key, creation_time) VALUES (?1, ?2, ?3)")
+            .unwrap();
+        for pre_key in pre_keys {
+            stmt.execute((
+                identity,
+                pre_key.to_bytes(),
+                SystemTime::now()
+                    .duration_since(UNIX_EPOCH)
+                    .unwrap()
+                    .as_secs(),
+            ))
+            .map_err(|_| Status::internal("failed to insert one time key"))?;
+        }
+        Ok(())
+    }
+
+    fn get_current_keys(
+        &self,
+        identity: &str,
+    ) -> tonic::Result<(VerifyingKey, proto::service::SignedPreKey)> {
+        eprintln!("Retrieving pre keys for user: {identity}");
+
+        let (ik, spk): (Vec<u8>, Vec<u8>) = self
```

Comment:
I don't like weird abbreviations.
```suggestion
        let (verifying_key, pre_key): (Vec<u8>, Vec<u8>) = self
```

Reply:
fixed